### DISCUSSION TITLE
It looks like last.fm scrobling not working on local mp3 played

### DISCUSSION THREAD

**Discussion by ha11oga11o** [rern/rAudio#1455] [General]
Hello all,

om trying rAudio to see will it fit my needs. Basically, i want RPI3 to be my music server just to play my whole mp3 collection from NAS to analog jack, to vintage sound system, with last.fm tracking. 

I somehow managed to create playlist and to play files, but theyre not reflected on my last.fm account. 
I authorized it with browser, all went fine without errors, but it simply not reflecting played songs to last.fm. 
I tried to see is there any log errors at /var/logs but nothing related. 

Can someone, please, point me out where to look, and is this option available at all?

I does working with another image, of another player. But that one i needed to do API manually and enter data.

Thank you in advance. 

Edit:

I verified under last.fm "connected applications" there is rAudio. Is it possible that it does not scribbling anything but airbplay, bluetooth, spitify and upnp as stated on settings?

Please forgive mine ignorance guys. 

Cheers

**Comment by rern** [comment]
Please try: `sed -i 's/\(mpd ]] \)||/\1\&\&/' /srv/http/bash/status-push.sh`

**Reply by ha11oga11o** [reply]
Hey, thanks for response. I gave up despite everything is better and easier than alternative. I have issues creating playlist library of 70+ k mp3 songs. It simply dont want to create it. I could do couple albums from USB drive. That works, but my NAS cant be synced to library. It frustrated me a lot, because it was flashing library bar for day or so, and then no cigar. It did nothing, absolutely nothing. Full rebuild and just missing files option. I click that it flashes thats it. 

I checked i could see folders and files under /mnt folder, and could browse with winscp from RPI3 itself. 

Many thnx and sorry for abandoning it for now, but it gets that family members did too much pressure to me to "just get back music play":)

Cheers

**Comment by ha11oga11o** [comment]
Hello,

sorry for bumping this topic up. But i finally manage to get rid of family demands and got spare RPI3 for usage. I reinstall rAudio on new card, did update, configuration and everything required to actually play couple albums from USB drive. All is fine, it works, but still no any reflection to last.fm webpage. No scrobbles at all. Can you please point me where is actual log to see whats happening there. I tried to find it with my limited knowledge but no cigar. 
I can confirm that player is linked on last.fm page, under applications. But it does not show last date/time usage as on their scrobbler is. It looks it never connected. 
I am not behind any firewalls too. 

I did copy paste that line above at putty interface, it did no output. And no any reflections to last.fm too. 

Many thnx in advance. 



rAudio 20231105
6.1.53-1 aarch64
Raspberry Pi 3B+ Rev 1.3
BCM2837B0 • 907 MiB
4 x Cortex-A53 @ 1.40 GHz

**Reply by rern** [reply]
Please try:
- Get the latest rAudio image (No need to run the command anymore)
- Features > Scrobbler - enable (also enable `Notify on scrobbling`)
- Play a song
- Scrobbler will notify after the end of the song

**Reply by ha11oga11o** [reply]
Im sadly saying scrobbling nor lst.fm webpage is not updating. I flashed sd with latest image. Is there any log location where error could be reflected? Can you please point me so i can extract it and check?
Again, on last.fm it states there's added application rAudio, but no date and time it actually last used. 

![Snimka zaslona 2023-11-07 085042](https://github.com/rern/rAudio/assets/1357309/b0ef91e8-0682-4ec1-b28b-adbfeb91430f)

Here is my last.fm account:

https://www.last.fm/user/ha11oga11o

Thank you in advance. 

rAudio 20231022
6.1.58-2 aarch64
Raspberry Pi 3B+ Rev 1.3
BCM2837B0 • 906 MiB
4 x Cortex-A53 @ 1.40 GHz
0.61 • 0.47 • 0.39

2023-11-07 • 07:55:21 UTC +00:00
24m since 2023-11-07 • 07:30

**Reply by ha11oga11o** [reply]
I have same issue after updating also:

rAudio 20231105
6.1.58-2 aarch64
Raspberry Pi 3B+ Rev 1.3
BCM2837B0 • 906 MiB
4 x Cortex-A53 @ 1.40 GHz
0.24 • 0.46 • 0.32
44.0 °C
2023-11-07 • 13:40:08 UTC +00:00
9m since 2023-11-07 • 13:30

**Reply by rern** [reply]
Please try:
- Addons > rAudio - Long-press Update/Install button
- Ok to UPDATE branch
- See whether it's fixed

**Reply by ha11oga11o** [reply]
No, it still dont work even with:

rAudio 20231111
6.1.58-2 aarch64
Raspberry Pi 3B+ Rev 1.3
BCM2837B0 • 906 MiB
4 x Cortex-A53 @ 1.40 GHz
1.09 • 1.20 • 0.74
41.9 °C
2023-11-13 • 10:50:35 UTC +00:00
10m since 2023-11-13 • 10:40

_

**Reply by ha11oga11o** [reply]
Any chance to tell me how to find logs related to this? I again did full sd card reflash, and still dont working at all. 

Thnx

**Reply by rern** [reply]
Try:
```sh
/srv/http/bash/scrobble.sh "cmd
Pink Floyd
Time
CMD ARTIST TITLE"
```

**Reply by ha11oga11o** [reply]
Yes, with that i got response from last.fm and in connected apps finally its registered that its connected.

Connected Applications

These are the applications that you have connected to your Last.fm account.

    rAudio

    Last used: Tuesday 14 Nov 2023, 8:23am


So, it seems it does not triggers when songs are actually playing? This is mpd log if it can help. I see no scrobbler info there at all. 

`mpd 0.23.14-1
# /srv/http/data/mpdconf/mpd.conf
audio_buffer_size       "8192"
bind_to_address         "0.0.0.0"
db_file                 "/srv/http/data/mpd/mpd.db"
max_output_buffer_size  "24576"
music_directory         "/mnt/MPD"
playlist_directory      "/srv/http/data/playlists"
state_file              "/var/lib/mpd/mpdstate"
user                    "mpd"
input {
	plugin  "curl"
}
decoder {
	plugin   "ffmpeg"
	enabled  "yes"
}
audio_output {
	type       "httpd"
	name       "Streaming"
	encoder    "flac"
	port       "8000"
	quality    "5.0"
	format     "44100:16:2"
	always_on  "yes"
}
audio_output {
	name           "On-board Headphones"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "PCM"
	mixer_device   "hw:0"
}

mpd.service- Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: active (running) since Tue 2023-11-14 07:08:01 UTC; 18min ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 82312 (mpd)
      Tasks: 7 (limit: 1006)
        CPU: 43.629s
     CGroup: /system.slice/mpd.service
             └─82312 /usr/bin/mpd --systemd /srv/http/data/mpdconf/mpd.conf

Nov 14 07:08:01 rAudio mpd[82312]: config: option 'quality' on line 6 was not recognized
Nov 14 07:08:01 rAudio systemd[1]: Started Music Player Daemon.
Nov 14 07:11:45 rAudio mpd[82312]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Nov 14 07:11:56 rAudio mpd[82312]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Nov 14 07:12:05 rAudio mpd[82312]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Nov 14 07:13:04 rAudio mpd[82312]: alsa_output: Decoder is too slow; playing silence to avoid xrun
Nov 14 07:16:22 rAudio mpd[82312]: player: played "NAS/NasData/Media/Music/Strana/Anthology/14 - Bad Company - Simple man (LP version 1995).mp3"
Nov 14 07:21:01 rAudio mpd[82312]: player: played "NAS/NasData/Media/Music/Strana/Anthology/14 - Bad Company - Simple man (LP version 1995).mp3"
Nov 14 07:25:41 rAudio mpd[82312]: player: played "NAS/NasData/Media/Music/Strana/Anthology/14 - Bad Company - Simple man (LP version 1995).mp3"`

Thanks for help!

**Reply by rern** [reply]
Please try the UPDATE branch again. It should be ok now.

**Reply by ha11oga11o** [reply]
After branch update all is working fine now. Got pop-up about scrobble, last.fm is populating correctly, without any additional characters. I would say thats it. I noticed bit strange thing, when skipping track it send it twice to last.fm. You can see that reflection on my lst.fm page. No errors or such on mpd log thou. 

Thanky you!


Its solved. Many thnx!

**Reply by ha11oga11o** [reply]
I think you should incorporate 80% of song listened to be published. This way its spamming last.fm if you press skip track. I manage to do multiples with skip button. I think, this needs more polishing?

![Last fm recent tracks](https://github.com/rern/rAudio/assets/1357309/58175982-fd03-47f2-a9ea-ab7b5b0256c6)


Spamming skip crashes player also:

`mpd 0.23.14-1
# /srv/http/data/mpdconf/mpd.conf
audio_buffer_size       "8192"
bind_to_address         "0.0.0.0"
db_file                 "/srv/http/data/mpd/mpd.db"
max_output_buffer_size  "24576"
music_directory         "/mnt/MPD"
playlist_directory      "/srv/http/data/playlists"
state_file              "/var/lib/mpd/mpdstate"
user                    "mpd"
input {
	plugin  "curl"
}
decoder {
	plugin   "ffmpeg"
	enabled  "yes"
}
audio_output {
	type       "httpd"
	name       "Streaming"
	encoder    "flac"
	port       "8000"
	quality    "5.0"
	format     "44100:16:2"
	always_on  "yes"
}
audio_output {
	name           "On-board Headphones"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "PCM"
	mixer_device   "hw:0"
}

mpd.service- Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: failed (Result: core-dump) since Tue 2023-11-14 11:07:19 UTC; 28min ago
   Duration: 45min 48.112s
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 530 ExecStart=/usr/bin/mpd --systemd /srv/http/data/mpdconf/mpd.conf (code=dumped, signal=SEGV)
   Main PID: 530 (code=dumped, signal=SEGV)
        CPU: 2min 16.570s

Nov 14 11:06:51 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/04 - las vegas.mp3"
Nov 14 11:06:52 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/04 - las vegas.mp3"
Nov 14 11:06:52 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/05 - deutsch nepal.mp3"
Nov 14 11:06:53 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/05 - deutsch nepal.mp3"
Nov 14 11:06:53 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/06 - utopia no. 1.mp3"
Nov 14 11:06:54 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/06 - utopia no. 1.mp3"
Nov 14 11:06:55 rAudio mpd[530]: player: played "NAS/NasData/Media/Music/Strana/Anthology/Amon Düül II/1973 - amon düül ii - utopia/07 - nasi goring - goreng ˇÚ.mp3"
Nov 14 11:07:19 rAudio systemd[1]: mpd.service: Main process exited, code=dumped, status=11/SEGV
Nov 14 11:07:19 rAudio systemd[1]: mpd.service: Failed with result 'core-dump'.
Nov 14 11:07:19 rAudio systemd[1]: mpd.service: Consumed 2min 16.570s CPU time.`
